python/tank/deploy/git_descriptor.py

```diff
@@ -259,8 +188,95 @@ def _find_latest_by_pattern(self, pattern):
             new_loc_dict,
             self._type
         )
-        
 
+    def _find_latest_tag_by_pattern(self, version_numbers, pattern):
```

Comment:
My initial approach to this method would be to just turn the pattern into a regular expression and search for matches against the start of the supplied version strings. Then use similar logic to find the most recent among the matches. Just curious what the advantage is to doing it this way or if I'm missing something obvious (highly likely). :)